vue-training/src/components/TechBox.vue

```diff
@@ -0,0 +1,95 @@
+<template lang="pug">
+  div.tech-box
+    img.tech-box-logo(
+      :src="tech.logo"
+      alt="tech logo"
+    )
+    h3.tech-box-title {{ tech.title }}
+    span.tech-box-launch Release: {{ tech.launch }}
+    p.tech-box-text
+      | {{ tech.description }}
+</template>
+<script>
+export default {
+  props: {
+    tech: {
+      type: Object,
+      required: true,
+      validator: value => value && value.title && value.launch && value.description && value.logo
```

Comment:
It's ok this, we are not used to doing this. When your component grows you will have an enormous function

Reply:
How can you easily validate that the object complies with a scheme?

Reply:
Using typescript for instance